### PR TITLE
python ovtf samples upgrade, yolov3 model conv with TF2.x

### PR DESCRIPTION
**Pull request by adisan3**
* Added multi input support for the existing ovtf samples.
* Tested the camera input for detection and classification.
* Created common_utils.py for the common code.
* Added conversion for yolov3 model with TF2.x.
* Added the pre and post processing for the yolov3 model input and outputs.
* Updated requirements.txt file for the samples.
* Moved the existing samples to TF1.x folder.
* Added the python object detection samples to the CI pipelines.
* Updated README.md.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 
+
 These examples demonstrate how to use **Intel<sup>®</sup> </sup> OpenVINO<sup>TM</sup> integration with Tensorflow** to recognize and detect objects in images and videos.
 
 ## Demos showcased in the examples
 
-* Classification demo uses Google's Inception v3 model to classify a given image or video.
-* Object detection demo uses YOLOv3 model to detect objects in a given image or video.
+* Classification demo uses Google's Inception v3 model to classify a given image, video, directory and camera input.
+* Object detection demo uses YOLOv3 daraknet model to detect objects in a given image, video, directory and camera input.
```
Comment (saipj) [inline examples/README.md:9]: spellcheck for darknet and directory to be edited to directory of images

File: examples/README.md [hunk 2]
```diff
@@ -61,27 +67,23 @@ bulletproof vest (466): 0.00535091
 
 In this case, we are using the default image of Admiral Grace Hopper. As you can see, the network correctly spots that she's wearing a military uniform, with a high score of 0.8.
 
-Next, try it out on your own image by passing the --image=argument, where argument is the path to your new image. You can provide either absolute or relative path in the argument  e.g.
-
+Next, try it out by passing the --input=argument, where argument is the path to your new image. You can provide either absolute or relative path to an image or video or directory.
+e.g.
```
Comment (saipj) [inline examples/README.md:71]: "path to your new image" to be changed to "path to your new input".

File: examples/README.md [hunk 3]
```diff
@@ -61,27 +67,23 @@ bulletproof vest (466): 0.00535091
...
 ```bash
-$ python3 examples/classification_sample.py --image=<absolute-or-relative-path-to-your-image>
+$ python3 examples/classification_sample.py --input=<absolute-or-relative-path-to-your-image>
```
Comment (saipj) [inline examples/README.md:73]: please change to path-to-your-input

File: examples/README.md [hunk 4]
```diff
@@ -93,19 +95,26 @@ If you add the new video (e.g, examples/data/people-detection.mp4) to the existi
...
 ## Python implementation for object detection
 
 For this example, we assume that you've already:
 
-* Installed TensorFlow on your system
+* Installed TensorFlow on your system (TF2.x)
 * Installed **Intel<sup>®</sup> </sup> OpenVINO<sup>TM</sup> integration with Tensorflow** on your system
 
+[**click-here**](https://github.com/openvinotoolkit/openvino_tensorflow/examples/TF_1_x/README.md) For the conversion of yolov3 darknet model using Tensorflow 1.15 and it's python object detection sample.
+
```
Comment (saipj) [inline examples/README.md:107]: pls modify "it's python object detection sample" to "object detection python sample to be used with this model"

File: examples/README.md [hunk 5]
```diff
@@ -134,28 +143,24 @@ output something similar as below:
 
 In this case, we're using the default image of Admiral Grace Hopper. As you can see, the network detects and draws the bounding box around the person correctly.
 
-Next, try it out on your own image by passing the --image=argument, where argument is the path to your new image. You can provide either absolute or relative path in the argument e.g.
+Next, try it out on your own image by passing the --input=argument, where argument is the path to your new image. You can provide either absolute or relative path in the argument e.g.
```
Comment (saipj) [inline examples/README.md:147]: Same changes for "image" to "input" in this README as well.

File: examples/TF_1_x/README.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+$ cd <path-to-openvino_tensorflow-repository>/examples/TF_1_x
+$ pip3 install -r requirements.txt
+```
+
+The TensorFlow Yolo v3 model used in this demo is not packaged in the repository because of its size. So, follow the instructions below to convert the model from DarkNet to TensorFlow and download the labels and weights to the `data` directory in your `cloned repo of openvino_tensorflow`:
+
+Please note: The instructions below should not be executed in an active virtual environment. The convert_yolov3.sh script activates a python virtual environment for conversion.
+
+```bash
+$ cd <path-to-openvino_tensorflow-repository>/examples
```
Comment (saipj) [inline examples/TF_1_x/README.md:40]: path to conversion script to changed to examples/TF_1_x

File: examples/TF_1_x/README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+```bash
+$ cd <path-to-openvino_tensorflow-repository>
+$ python3 examples/TF_1_x/object_detection_sample.py
+```
+
+This uses the default example image that comes with this repository, and should
+output something similar as below:
+
+<p align="left">
+  <img src="../examples/TF_1_x/data/detections.jpg" width="200" height="200"
```
Comment (saipj) [inline examples/TF_1_x/README.md:61]: we dont need new data folder in TF_1_x. Existing data folder in examples should be leveraged.

File: examples/TF_1_x/README.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+</p>
+
+In this case, we're using the default image of Admiral Grace Hopper. As you can see, the network detects and draws the bounding box around the person correctly.
+
+Next, try it out on your own image by passing the --image=argument, where argument is the path to your new image. You can provide either absolute or relative path in the argument e.g.
+
+```bash
+$ python3 examples/TF_1_x/object_detection_sample.py --image=<absolute-or-relative-path-to-your-image>
```
Comment (saipj) [inline examples/TF_1_x/README.md:69]: please verify all lines in README to change "image" to "input"

File: examples/TF_1_x/README.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+```bash
+$ python3 examples/TF_1_x/object_detection_sample.py --help
+```
+To run object detection sample with video input, follow the instructions below:
+```bash
+$ pip3 install opencv-python
+$ cd <path-to-openvino_tensorflow-repository>
+$ python3 examples/TF_1_x/object_detection_sample_video.py
+```
+Next, try it out on your own video file by passing the --input=argument, where argument is the path to your input video. You can provide either absolute or relative path in the argument  e.g.
```
Comment (saipj) [inline examples/TF_1_x/README.md:88]: these need to be removed since  *_video.py script no longer valid.

File: examples/TF_1_x/README.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+```bash
+$ python3 examples/TF_1_x/object_detection_sample_video.py --input=<absolute-or-relative-path-to-your-video-file>
```
Comment (saipj) [inline examples/TF_1_x/README.md:91]: same here. no video.

File: examples/TF_1_x/README.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+```
+If you add the new video (e.g, examples/TF_1_x/data/people-detection.mp4) to the existing data directory in the openvino_tensorflow repository, it will look like this:
+
```
Comment (saipj) [inline examples/TF_1_x/README.md:94]: Please update. no TF_1_x/data folder.

File: examples/TF_1_x/README.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+```bash
+$ python3 examples/TF_1_x/object_detection_sample_video.py --input=examples/TF_1_x/data/people-detection.mp4
```
Comment (saipj) [inline examples/TF_1_x/README.md:96]: no *_video.py script

File: examples/TF_1_x/README.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+To try on the yolo_v3_160 model for faster inference follow the below steps
+
+Please note: The instructions below should not be executed in an active virtual environment. The convert_yolov3_160.sh script activates a python virtual environment for conversion.
+
+```bash
+$ cd <path-to-openvino_tensorflow-repository>/examples
```
Comment (saipj) [inline examples/TF_1_x/README.md:104]: script moved to TF_1_x folder. Please update.

File: examples/TF_1_x/README.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+$ chmod +x convert_yolov3_160.sh
+$ ./convert_yolov3_160.sh
+```
+
+Run the object detection example using the instructions below:
+
+```bash
+$ cd <path-to-openvino_tensorflow-repository>
+$ python3 examples/TF_1_x/object_detection_sample_video.py --input_height 160 --input_width 160 --graph "examples/TF_1_x/data/yolo_v3_160.pb" --input_layer "inputs" --output_layer "output_boxes" --labels "examples/TF_1_x/data/coco.names"
```
Comment (saipj) [inline examples/TF_1_x/README.md:113]: no video sample. please update.

File: examples/TF_1_x/common_utils.py [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+def load_graph(model_file):
+    graph = tf.Graph()
+    graph_def = tf.compat.v1.GraphDef()
+    assert os.path.exists(model_file), "Could not find model path"
+    with open(model_file, "rb") as f:
+        graph_def.ParseFromString(f.read())
+    with graph.as_default():
+        tf.import_graph_def(graph_def)
+
+    return graph
```
Comment (saipj) [inline examples/TF_1_x/common_utils.py:36]: isn't the APIs from this utility file same across TF 1.x and TF 2.x implementations? if so, we can have just one common_utils.py as default in examples folder instead of two duplicate copies.

File: examples/TF_1_x/common_utils.py [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+        for i in images:
+            image_path = os.path.join(input_path, i)
+            if imghdr.what(image_path) == None:
+                assert False, "Input directory contains non image files"
+        return "directory"
+    elif os.path.isfile(input_path):
+        if imghdr.what(input_path) != None:
+            return "image"
+        elif input_path.rsplit('.', 1)[1] in ['mp4', 'avi']:
+            return "video"
```
Comment (saipj) [inline examples/TF_1_x/common_utils.py:24]: Pls add else and return "please provide correct input and list of image/video formats supported here".

File: examples/TF_1_x/object_detection_sample.py [hunk 17]
```diff
@@ -164,10 +156,18 @@ def non_max_suppression(predictions_with_boxes,
...
+    for l in proto_as_ascii_lines:
+        label.append(l.rstrip())
+    return label
+
+
 if __name__ == "__main__":
-    input_file = "examples/data/people-detection.mp4"
-    model_file = "examples/data/yolo_v3_darknet.pb"
-    label_file = "examples/data/coco.names"
+    input_file = "data/grace_hopper.jpg"
```
Comment (saipj) [inline examples/TF_1_x/object_detection_sample.py:168]: not consistent with README instructions. No additional data folder needed in TF_1_x folder.

File: examples/TF_1_x/object_detection_sample.py [hunk 18]
```diff
@@ -131,8 +121,10 @@ def non_max_suppression(predictions_with_boxes,
     result = {}
     for i, image_pred in enumerate(predictions):
         shape = image_pred.shape
-        non_zero_idxs = np.nonzero(image_pred)
-        image_pred = image_pred[non_zero_idxs]
+        temp = image_pred
+        sum_t = np.sum(temp, axis=1)
+        non_zero_idx = sum_t != 0
+        image_pred = image_pred[non_zero_idx, :]
         image_pred = image_pred.reshape(-1, shape[-1])
```
Comment (saipj) [inline examples/TF_1_x/object_detection_sample.py:128]: not sure why 4 lines of code is added instead of existing 2 lines? can you explain?

File: examples/TF_1_x/README.md (hunk 13 above)
Comment (saipj) [inline examples/TF_1_x/README.md:104]: conversion script is moved to examples/TF_1_x folder.

File: examples/keras_to_tensorflow.patch [hunk 19]
```diff
@@ -0,0 +1,22 @@
+--- tensorflow-yolo-v3/tools/model_converter/keras_to_tensorflow.py	2021-08-17 04:17:22.186256673 +0530
++++ keras_to_tensorflow_modified.py	2021-08-17 04:14:43.686249508 +0530
+@@ -22,7 +22,9 @@
+ 
+ sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..'))
+ from common.utils import get_custom_objects
+-
++# tf.enable_eager_execution()
```
Comment (sspintel) [inline examples/keras_to_tensorflow.patch:8]: If this line is commented out in the patch, it doesn't have to be a part of the patch in the first place, right?

**Review by sspintel — APPROVED** [review]
Changes look good to me
